src/Domains/Connectors/RainForest/Workflows/Activities/ImportProductActivity.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Kanvas\Connectors\RainForest\Workflows\Activities;
+
+use Baka\Contracts\AppInterface;
+use Illuminate\Support\Str;
+use Kanvas\Companies\Models\CompaniesBranches;
+use Kanvas\Connectors\RainForest\Repositories\ProductRepository;
+use Kanvas\Inventory\Channels\Models\Channels;
+use Kanvas\Inventory\Importer\Jobs\ProductImporterJob;
+use Kanvas\Inventory\Regions\Models\Regions;
+use Kanvas\Users\Models\Users;
+use Workflow\Activity;
+
+class ImportProductActivity extends Activity
+{
+    public function execute(AppInterface $app, Users $users, CompaniesBranches $companyBranch, Regions $region, string $search)
+    {
+        try {
+            $warehouse = $region->warehouses()->where('is_default', true)->first();
```

Comment:
@FredPeal move the logic to import to a action so you can use it in other places if needed